### PR TITLE
Address final stylecop warnings in Skills/Solutions nuget

### PR DESCRIPTION
**Pull request by darrenj**
In preparation for formal libraries to be published I've gone through and addressed all outstanding stylecop issues to achieve zero warnings.

Directory.Build.Props:
- Supressing NU1701 keyvault error
- Warnings as Errors enabled

SkillDialog
- Put Action mapping code back in and addressed bug where exception was thrown if it could not find the action which will mean it works for both grouped dispatch and single action dispatch scenarios

General
- ConfigureAwait everywhere
- Async postfix to async methods

### CODE REVIEW

File: lib/csharp/microsoft.bot.builder.skills/Microsoft.Bot.Builder.Skills/SkillDialog.cs [hunk 1]
```diff
@@ -87,38 +88,39 @@ protected override async Task<DialogTurnResult> OnBeginDialogAsync(DialogContext
...
-            //     else
-            //     {
-            //         // Loosening checks for current Dispatch evaluation, TODO - Review
-            //         // throw new ArgumentException($"Passed Action ({actionName}) could not be found within the {_skillManifest.Id} skill manifest action definition.");
-            //     }
-            // }
+            var actionName = options as string;
+            if (actionName == null)
+            {
+                throw new ArgumentException("SkillDialog requires an Action in order to be able to identify which Action within a skill to invoke.");
```
Comment (lauren-mills) [inline lib/csharp/microsoft.bot.builder.skills/Microsoft.Bot.Builder.Skills/SkillDialog.cs:93]: Seems like this will still break without an action, right?

Comment (darrenj) [reply lib/csharp/microsoft.bot.builder.skills/Microsoft.Bot.Builder.Skills/SkillDialog.cs:93]: Correct but you are passing an "action" - the dispatch intent from maindialog so you won't (shouldn't) hit that

**Comment by darrenj** [comment]
@lzc850612  updated to the new nuget repository for the protocol libraries to resolve PR failure. https://botbuilder.myget.org/F/experimental/api/v3/index.json

File: lib/csharp/microsoft.bot.builder.skills/Microsoft.Bot.Builder.Skills/SkillDialog.cs (hunk 1 above)
Comment (lauren-mills) [reply lib/csharp/microsoft.bot.builder.skills/Microsoft.Bot.Builder.Skills/SkillDialog.cs:93]: Ok, got it

**Review by lauren-mills — APPROVED** [review]

**Review by lzc850612 — APPROVED** [review]

**Pull request merged**